ledger-core/virtual/integration/mock/typepublishchecker.go

```diff
@@ -0,0 +1,468 @@
+		count         atomickit.Int
+		expectedCount int
+		handler       VCallRequestHandler
+	}
+	VCallResult struct {
+		touched       bool
+		count         atomickit.Int
+		expectedCount int
+		handler       VCallResultHandler
+	}
+	VDelegatedCallRequest struct {
+		touched       bool
+		count         atomickit.Int
+		expectedCount int
+		handler       VDelegatedCallRequestHandler
+	}
+	VDelegatedCallResponse struct {
+		touched       bool
+		count         atomickit.Int
+		expectedCount int
+		handler       VDelegatedCallResponseHandler
+	}
+	VDelegatedRequestFinished struct {
+		touched       bool
+		count         atomickit.Int
+		expectedCount int
+		handler       VDelegatedRequestFinishedHandler
+	}
+	VStateRequest struct {
+		touched       bool
+		count         atomickit.Int
+		expectedCount int
+		handler       VStateRequestHandler
+	}
+	VStateReport struct {
+		touched       bool
+		count         atomickit.Int
+		expectedCount int
+		handler       VStateReportHandler
+	}
+
+	BaseMessage struct {
+		handler func(message *message.Message)
+	}
+}
+
+type TypePublishChecker struct {
+	t             minimock.Tester
+	ctx           context.Context
+	defaultResend bool
+	resend        func(ctx context.Context, msg *message.Message)
+
+	Handlers TypePublishCheckerHandlers
+
+	VCallRequest              PubVCallRequestMock
+	VCallResult               PubVCallResultMock
+	VDelegatedCallRequest     PubVDelegatedCallRequestMock
+	VDelegatedCallResponse    PubVDelegatedCallResponseMock
+	VDelegatedRequestFinished PubVDelegatedRequestFinishedMock
+	VStateRequest             PubVStateRequestMock
+	VStateReport              PubVStateReportMock
+}
+
+func NewTypePublishChecker(ctx context.Context, t minimock.Tester, sender Sender) *TypePublishChecker {
+	checker := &TypePublishChecker{
+		t:             t,
+		ctx:           ctx,
+		defaultResend: false,
+		resend:        sender.SendMessage,
+	}
+
+	checker.VCallRequest = PubVCallRequestMock{parent: checker}
+	checker.VCallResult = PubVCallResultMock{parent: checker}
+	checker.VDelegatedCallRequest = PubVDelegatedCallRequestMock{parent: checker}
+	checker.VDelegatedCallResponse = PubVDelegatedCallResponseMock{parent: checker}
+	checker.VDelegatedRequestFinished = PubVDelegatedRequestFinishedMock{parent: checker}
+	checker.VStateRequest = PubVStateRequestMock{parent: checker}
+	checker.VStateReport = PubVStateReportMock{parent: checker}
+
+	if controller, ok := t.(minimock.MockController); ok {
+		controller.RegisterMocker(checker)
+	}
+
+	return checker
+}
+
+func (p *TypePublishChecker) CheckMessages(topic string, messages ...*message.Message) error {
+	for _, msg := range messages {
+		p.checkMessage(p.ctx, msg)
+	}
+
+	return nil
+}
+
+func (p *TypePublishChecker) checkMessage(ctx context.Context, msg *message.Message) {
+	basePayload, err := payload.UnmarshalFromMeta(msg.Payload)
+	if err != nil {
+		return
+	}
+
+	var resend bool
+
+	switch payload := basePayload.(type) {
+	case *payload.VCallRequest:
```

Comment:
this looks like a registry map with one check, because all the checks are indentical

Reply:
What do you mean by registry map?

Reply:
I can't see any other ways for maps except using reflect here